src/lib.rs

```diff
@@ -3918,6 +4021,92 @@ mod tests {
         }
     }
 
+    mod dst_layout {
```

Comment:
These tests should probably also exercise calling `extend` with a maximum alignment argument.